tensorflow/python/autograph/operators/py_builtins.py

```diff
@@ -322,10 +334,30 @@ def print_(*objects, **kwargs):
   else:
     _py_print(*objects, **kwargs)
 
-
 def _py_print(*objects, **kwargs):
   print(*objects, **kwargs)
 
+def max_(*args, **kwargs):
+  if any(tensor_util.is_tf_type(s) for s in args):
+    return _tf_max(*args, **kwargs)
+  return _py_max(*args, **kwargs)
+
+def _tf_max(*args, **kwargs):
+  if len(kwargs):
+    kwargs_tuple = tuple(set(kwargs.keys()))
+    raise ValueError('These keyword arguments are ' 
+                     'currently not supported: {}'.format(kwargs_tuple))
+  elif len(args) == 1:
+    # TODO (bhack) Do we want the input Tensor to be rank==1 or scalar?
+    # if not something like [[1, 2][3,4]] it will give an unexpected result
+    return  math_ops.reduce_max(*args, axis=0)
+  else:
+    if all(_tf_tensor_is_scalar(arg) for arg in args):
```

Comment:
This is a bit more complicated, and we can't do it with a helper function because of dynamic shapes. We need something like this (approximately):

```
if any(t.shape.rank is None for t in args):
  # Dynamic shape
  with ops.control_dependencies(
    [
      control_flow_ops.Assert(math_ops.equal(array_ops.rank(s), 1), ["max requires all inputs to be scalar"])
      for s in args
    ]):
    return math_ops.reduce_max(array_ops.stack(args), axis=0)
else:
  # Static ranks
  if any(t.shape.rank != 1):
    raise ValueError("max requires all inputs to be scalar")
  return math_ops.reduce_max(array_ops.stack(args), axis=0)
```

Unfortunately it's a bit duplicated because of the context manager.

Reply:
I don't know the context manager internals  but I trust you that it need to be done internally in the same function.